python/ray/tests/test_gcs_fault_tolerance.py

```diff
@@ -214,6 +215,71 @@ def test_gcs_client_reconnect(ray_start_regular_with_external_redis, auto_reconn
         assert gcs_client.internal_kv_get(b"a", None) == b"b"
 
 
+@pytest.mark.parametrize(
+    "ray_start_regular_with_external_redis",
+    [
+        {
+            **generate_system_config_map(
+                num_heartbeats_timeout=20, gcs_rpc_server_reconnect_timeout_s=3600
+            ),
+            "namespace": "actor",
+        }
+    ],
+    indirect=True,
+)
+def test_actor_workloads(ray_start_regular_with_external_redis):
+    """This test cover the case to create actor while gcs is down
+    and also make sure existing actor continue to work even when
+    GCS is down.
+    """
+
+    @ray.remote
+    class Counter:
+        def r(self, v):
+            return v
+
+    c = Counter.remote()
+    r = ray.get(c.r.remote(10))
+    assert r == 10
+
+    print("GCS is killed")
+    ray.worker._global_node.kill_gcs_server()
+
+    print("Start to create a new actor")
+    cc = Counter.remote()
+    with pytest.raises(ray.exceptions.GetTimeoutError):
+        ray.get(cc.r.remote(10), timeout=5)
+
+    assert ray.get(c.r.remote(10)) == 10
```

Comment:
> When GCS went down and there is a background request to GCS (which may be hanging), core worker's main event loop (ray.get()?) still works.

@mwtian  I covered it here.

Reply:
I mean can we call `ray.get(cc.r.remote(10))` on a separate thread without timeout? Just to make sure concurrent calls are working, even when another call is hanging..